File: ecom_front/src/types/Product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  brand: string;
  price: number;
  category?: string;
  releaseDate: Date;
  available?: boolean;
  quantity: number;
  imageName?: string;
  imageType?: string;
  image?: string; // Base64 encoded image data for frontend display
}

File: ecom/src/main/java/org/example/ecom/model/Product.java
package org.example.ecom.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Product {

    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String description;
    private String brand;
    private int price;
    private String category;
    private Date releaseDate;
    private boolean available;
    private int quantity;

/*    private String imageName;
    private String imageType;

    @Lob
    private byte[] image;*/

}
